src/gui/plugins/tape_measure/TapeMeasure.cc

```diff
@@ -0,0 +1,282 @@
+}
+
+/////////////////////////////////////////////////
+double TapeMeasure::Distance()
+{
+  return this->dataPtr->distance;
+}
+
+/////////////////////////////////////////////////
+void TapeMeasure::DeleteMarker(int _id)
+{
+  if (this->dataPtr->placedMarkers.find(_id) ==
+      this->dataPtr->placedMarkers.end())
+    return;
+
+  // Delete the previously created marker
+  ignition::msgs::Marker markerMsg;
+  markerMsg.set_ns(this->dataPtr->ns);
+  markerMsg.set_id(_id);
+  markerMsg.set_action(ignition::msgs::Marker::DELETE_MARKER);
+  this->dataPtr->node.Request("/marker", markerMsg);
+  this->dataPtr->placedMarkers.erase(_id);
+}
+
+/////////////////////////////////////////////////
+void TapeMeasure::DrawPoint(int _id,
+    ignition::math::Vector3d &_point, ignition::math::Vector4d &_color)
+{
+  this->DeleteMarker(_id);
+
+  ignition::msgs::Marker markerMsg;
+  markerMsg.set_ns(this->dataPtr->ns);
+  markerMsg.set_id(_id);
+  markerMsg.set_action(ignition::msgs::Marker::ADD_MODIFY);
+  markerMsg.set_type(ignition::msgs::Marker::SPHERE);
+  ignition::msgs::Set(markerMsg.mutable_scale(),
+    ignition::math::Vector3d(0.1, 0.1, 0.1));
+  markerMsg.mutable_material()->mutable_ambient()->set_r(_color[0]);
+  markerMsg.mutable_material()->mutable_ambient()->set_g(_color[1]);
+  markerMsg.mutable_material()->mutable_ambient()->set_b(_color[2]);
+  markerMsg.mutable_material()->mutable_ambient()->set_a(_color[3]);
+  markerMsg.mutable_material()->mutable_diffuse()->set_r(_color[0]);
+  markerMsg.mutable_material()->mutable_diffuse()->set_g(_color[1]);
+  markerMsg.mutable_material()->mutable_diffuse()->set_b(_color[2]);
+  markerMsg.mutable_material()->mutable_diffuse()->set_a(_color[3]);
+  ignition::msgs::Set(markerMsg.mutable_pose(),
+    ignition::math::Pose3d(_point.X(), _point.Y(), _point.Z(), 0, 0, 0));
+
+  this->dataPtr->node.Request("/marker", markerMsg);
+  this->dataPtr->placedMarkers.insert(_id);
+}
+
+/////////////////////////////////////////////////
+void TapeMeasure::DrawLine(int _id, ignition::math::Vector3d &_startPoint,
+    ignition::math::Vector3d &_endPoint, ignition::math::Vector4d &_color)
+{
+  this->DeleteMarker(_id);
+
+  ignition::msgs::Marker markerMsg;
+  markerMsg.set_ns(this->dataPtr->ns);
+  markerMsg.set_id(_id);
+  markerMsg.set_action(ignition::msgs::Marker::ADD_MODIFY);
+  markerMsg.set_type(ignition::msgs::Marker::LINE_LIST);
+  markerMsg.mutable_material()->mutable_ambient()->set_r(_color[0]);
+  markerMsg.mutable_material()->mutable_ambient()->set_g(_color[1]);
+  markerMsg.mutable_material()->mutable_ambient()->set_b(_color[2]);
+  markerMsg.mutable_material()->mutable_ambient()->set_a(_color[3]);
+  markerMsg.mutable_material()->mutable_diffuse()->set_r(_color[0]);
+  markerMsg.mutable_material()->mutable_diffuse()->set_g(_color[1]);
+  markerMsg.mutable_material()->mutable_diffuse()->set_b(_color[2]);
+  markerMsg.mutable_material()->mutable_diffuse()->set_a(_color[3]);
```

Comment:
How about using `ignition::math::Color` for all the colors?

Then you can also use [ignition::msgs::Set](https://github.com/ignitionrobotics/ign-msgs/blob/1c3dfd91463090f7503d710176851c3c722977b5/include/ignition/msgs/Utility.hh#L337-L341) to set the message fields from the colors.

Reply:
dda9038aef5fc0e953aadaea329fca4f37d20695